rtc_base/task_queue_stdlib.cc

```diff
@@ -297,19 +286,22 @@ void TaskQueue::Impl::ThreadMain(void* context) {
     if (me->thread_should_quit_)
       break;
 
-    std::unique_lock<std::mutex> flag_lock(me->flag_lock_);
-    if (Microseconds() == sleep_time)
-      me->flag_notify_.wait(flag_lock);
-    else
-      me->flag_notify_.wait_for(flag_lock, sleep_time);
+    if (Microseconds{} == sleep_time_us) {
+      me->flag_notify_.Wait(rtc::Event::kForever);
```

Comment:
>kForever [](start = 40, length = 8)

Won't this allow other kind of dead locks to happen?

Reply:
How so? The only issue with the current code is that APC calls only fire if the thread enters an alertable state, and the synchronization primitive used doesn't do that. So removing the APC code and using another mechanism to do the same synchronization fixes the issue. This `Wait(kForever)` is the thread waiting for work to be submitted to it, it's central to the task queue functioning.

Also this is probably not visible from that review but I am essentially copy/pasting the code which has been submitted to the Google repository (see [this change](https://webrtc.googlesource.com/src/+/22027b92d6a3b4022ac744b3a89e0e0dcf9d0b49)) a few months after being pushed to this one, which for some reasons I ignore is quite a bit different from the current code. It has been reviewed on Google's side and I also tested it myself. The change brings closer the two repositories; there are still differences, but the interface for the task queue is also changed in later milestones which doesn't help comparing.